Readme.md

```diff
@@ -37,48 +37,12 @@ NumberList ::= number (sep number)*
 Token input using BNF
 `1 2 3`
 
-Token output
-```
-Some(
-    Token {
-        name: Some(
-            "NumberList",
-        ),
-        match: "1 2 3",
-        range: 0..5,
-        children: [
-            Token {
-                name: Some(
-                    "number",
-                ),
-                match: "1",
-                range: 0..1,
-            },
-            Token {
-                name: Some(
-                    "number",
-                ),
-                match: "2",
-                range: 2..3,
-            },
-            Token {
-                name: Some(
-                    "number",
-                ),
-                match: "3",
-                range: 4..5,
-            },
-        ],
-        ..
-    },
-)
-```
-To apply the rules to the `lexer` you simpily do
+The BNF rules is what defines the syntax of the stuff thats being passed in. The lexer rules tell the lexer what to do with those things that are being passed in from the from the rules list in the lexer, as seen below.
 ```rust
 lexer.add_rule_for_names(vec!["sep"], CullStrategy::DeleteAll);
 ```
 
-This is what the tokenized tree would look like before applying the rules
+Below is what the tokenized tree is what it would look like before applying the rules and the second tree is what it would look like after the rules were applied. You can see that the seps were taken out of the tree making it easier to read.
```

Comment:
Explaining what both are above the first one is a little confusing - you might want to split it so there's an explanation above each one.